src/Engine/Check/Database/RequirePrimaryKey.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cadfael\Engine\Check\Database;
+
+use Cadfael\Engine\Check;
+use Cadfael\Engine\Entity\Database;
+use Cadfael\Engine\Exception\MySQL\UnknownVersion;
+use Cadfael\Engine\Report;
+
+/**
+ * Class RequirePrimaryKey
+ * @package Cadfael\Engine\Check\Schema
+ */
+class RequirePrimaryKey implements Check
+{
+    public function supports($entity): bool
+    {
+        return $entity instanceof Database;
+    }
+
+    public function run($entity): ?Report
+    {
+        try {
+            $version = $entity->getVersion();
+        } catch (UnknownVersion $e) {
+            return new Report(
+                $this,
+                $entity,
+                Report::STATUS_CONCERN,
+                [
+                    $e->getMessage(),
+                    "This makes us nervous."
+                ]
+            );
+        }
+
+        $variables = $entity->getVariables();
+        $opt_name = 'sql_require_primary_key';
+        $require_pk_disabled = (!isset($variables[$opt_name]) || $variables[$opt_name] !== 'ON');
+        if (version_compare($entity->getVersion(), '8.0.0', '>=') && $require_pk_disabled) {
```

Comment:
FYI this was introduced in `8.0.13` (my bad for not checking [here](https://dev.mysql.com/doc/mysqld-version-reference/en/optvar-changes-8-0.html)) so if you want another hacktober MR, this is a quick one.

Reply:
You can run it against your test environment database without `--performance_schema`. Certain checks won't run (like badly performing queries and inactive users) and others around capacity planning won't give relevant warnings, but the design checks will return valuable results.